components/script/dom/webgl2renderingcontext.rs

```diff
@@ -208,20 +251,44 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
     }
 
     #[allow(unsafe_code)]
-    /// https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
     unsafe fn BufferData(&self, cx: *mut JSContext, target: u32, data: *mut JSObject, usage: u32) -> Fallible<()> {
-        self.base.BufferData(cx, target, data, usage)
+        let bound_buffer = match target {
+            constants::ARRAY_BUFFER => self.base.bound_buffer_array(),
+            constants::ELEMENT_ARRAY_BUFFER => self.base.bound_buffer_element_array(),
+            constants::UNIFORM_BUFFER => self.bound_uniform_buffer.get(),
+            constants::TRANSFORM_FEEDBACK_BUFFER => self.bound_transform_feedback_buffer.get(),
+            _ => return Ok(self.webgl_error(InvalidEnum)),
+        };
+
+        self.base.buffer_data(cx, target, bound_buffer, data, usage)
     }
 
-    /// https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.5
     fn BufferData_(&self, target: u32, size: i64, usage: u32) -> Fallible<()> {
-        self.base.BufferData_(target, size, usage)
+        let bound_buffer = match target {
```

Comment:
It would be nice if we could have a single helper function that returned the binding point (not the bound buffer) given the target enum.